src/Chunk.ts

```diff
@@ -414,7 +414,7 @@ export default class Chunk {
 			if (module.preserveSignature === 'allow-extension') {
 				const canPreserveExports = this.canPreserveModuleExports(module);
 
-				if (canPreserveExports) {
+				if (canPreserveExports && !module.chunkFileNames.size && !module.chunkNames.length) {
```

Comment:
While having entries in module.chunkFileNames is definitely a condition to create a separate chunk, I am not 100% sure about module.chunkNames. Essentially, chunkNames are just hints how to build the name. What I would have expected is `module.chunkNames.every(name => name.!isUserDefined)`. This is false if any of the chunk names is set in package.json. On the other hand, maybe your fix is needed for #5907?

Reply:
For #5907, the condition `module.chunkFileNames.size` could solve it. I added the condition `!module.chunkNames.length` for #5906. You're right, for this specific case, we should check `isUserDefined`.